API-strategie-modules/transport-security/transport-security.md

```diff
@@ -11,22 +11,29 @@ In order to meet the complete security objectives, every implementer MUST also a
 
-<div class="rule" id="api-11">
-  <p class="rulelab"><strong>API-11</strong>: Secure connections using TLS</p>
+<span id="api-11"></span>
+<div class="rule" id="/transport/TLS">
+  <p class="rulelab"><strong>/transport/TLS</strong>: Secure connections using TLS</p>
   <p>Secure connections using TLS following the latest NCSC guidelines [[NCSC.TLS]].</p>
   <p>Since the connection is always secured, the access method can be straightforward. This allows the application of basic access tokens instead of encrypted access tokens.</p>
 </div>
 
 Even when using TLS-based secured connections information in URIs is not secured. URIs can be cached and logged outside of the servers controlled by clients and servers. Any information contained in them should therfor be considered readable by anyone with access to the netwerk being used (in case of the internet the whole world) and MUST NOT contain any sensitive information. Neither client secrets used for authentication, privacy sensitive informations suchs as BSNs nor any other information which should not be shared. Be aware that queries (anything after the '?' in a URI) are also part of an URI.
 
-<div class="rule" id="api-58">
-  <p class="rulelab"><strong>API-58</strong>: No sensitive information in URIs</p>
+<span id="api-58"></span>
+<div class="rule" id="/transport/no-sensitive-URIs">
+  <p class="rulelab"><strong>/transport/no-sensitive-URIs</strong>: No sensitive information in URIs</p>
   <p>Do not put any sensitive information in URIs</p>
   <p>Even when the connection is secure URIs can be cached and logged, in systems outside the control of client and/or server.</p>
 </div>
 
+**How to test**
+Sensitive information URIS is not machine testable and therfor not part of automated tests. It should be part of any security audit performed by human experts. 
+
+The usage of TLS is machine testable. The test is designed for maximum automation. To test, adherence to NCSC reccomendations should be tested. The serverside is what will be tested, only control over the server is assumed for testing. A testing client will be employed to test adherence of the server. Supporting any protocols, algorithms, key sizes, options or ciphers dat are deemed insufficient or phase out by NCSC will lead to failure on the automated test. Both positive and negative scenario's are part of the test. Testing that a subset of good and sufficient reccomendations are supported and testing that phase out and insufficient reccomendations are not. A manual exception to the automated test results can be made when phase out reccomendations are supported. The API provider will have to provide clear documentation for the phase out schedule.  
```

Comment:
Wat doen we wanneer het NCSC met nieuwe aanbevelingen komt>

Een regel gaat van `good` naar `phase out`. Denk dat het goed is om x-tijd te geven om daar aan te voldoen. Maw: wat is uitfaseren, hoeveel tijd mag dat kosten?

> A manual exception to the automated test results can be made when phase out reccomendations are supported. The API provider will have to provide clear documentation for the phase out schedule.

Als we een tijdslimiet stellen aan uitfaseren is een handmatige uitzondering wellicht  niet meer nodig.

Zo niet, dan moet z'n uitzondering wel kenbaar gemaakt worden aan de automatisch test. Een geschikte plaats is de OAS in een bepaald veld.